src/interpreter/index.ts

```diff
@@ -252,63 +226,183 @@ function translateStep<T, I>(
         case "smallerEqual":
         case "subtract":
         case "unequal":
-            return interpreteBinaryOperator(step, context, next)
+            return interpreteBinaryOperator(value, step, context)
         case "if":
-            return interpreteIf(step, context, next)
+            return interpreteIf(value, step, context)
         case "switch":
-            return interpreteSwitch(step, context, next)
+            return interpreteSwitch(value, step, context)
         case "getVariable":
-            return interpreteGetVariable(step, next)
+            return interpreteGetVariable(value, step)
         case "setVariable":
-            return interpreteSetVariable(step, context, next)
-        case "return":
-            return interpreteReturn()
+            return interpreteSetVariable(value, step, context)
         case "random":
-            return interpreteRandom(step, context, next)
-        case "null":
-            return interpreteNull()
+            return interpreteRandom(value, step, context)
+    }
+    return {} as Value<T>[]
+}
+
+function interpreteRandom<T, A, I>(
+    value: Value<T>[],
+    step: AbstractParsedRandom<I>,
+    context: InterpretionContext<T, I>
+): Value<T>[] {
+    const newValue = [] as Value<T>[]
+
+    const stepProb: number[] = []
+    step.probabilities.forEach((curr, i) =>
+        i == 0 ? stepProb.push(curr) : stepProb.push(curr + stepProb[stepProb.length - 1])
+    )
+    value.forEach((v) => {
```

Comment:
Generell würde ich immer `for(const x of ...)` bevorzugen. Ist schneller und hat mehr features (`return` direkt aus der scheife, und `break`)